### PR TITLE
#292 Add support for automatic tailwind config merge

### PR DESCRIPTION
**Pull request by wigman**

### CODE REVIEW

File: src/Observer/RegisterModuleForHyvaConfig.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
```
Comment (rajeev-k-tomy) [inline src/Observer/RegisterModuleForHyvaConfig.php:2]: i dont think we need this phpdoc here?

File: src/Observer/RegisterModuleForHyvaConfig.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        $this->componentRegistrar = $componentRegistrar;
+    }
+
+    public function execute(Observer $event)
+    {
+        $config = $event->getData('config');
+        $extensions = $config->hasData('extensions') ? $config->getData('extensions') : [];
+
+        $moduleName = implode('_', array_slice(explode('\\', __CLASS__), 0, 2));
```
Comment (rajeev-k-tomy) [inline src/Observer/RegisterModuleForHyvaConfig.php:34]: Nice move. Eventhough people change the module name, this still works 👍

File: src/view/frontend/tailwind/tailwind.config.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Since Hyvä Themes 1.1.14 this file is automatically
+ * merged into your Hyvä Theme Tailwind config
+ *
+ * See https://docs.hyva.io/hyva-themes/compatibility-modules/tailwind-config-merging.html
+ *
+ **/
+
+module.exports = {
+    purge: {
```
Comment (rajeev-k-tomy) [inline src/view/frontend/tailwind/tailwind.config.js:10]: Two concerns.

1. Need to include phtml files also in the purge list.
2. What if the theme use version 3? then `purge.content` wont work. Only `content` path exist there I believe.

**Review by rajeev-k-tomy — CHANGES_REQUESTED** [review]
Please have a look on my comments

File: src/view/frontend/tailwind/tailwind.config.js (hunk 3 above)
Comment (wigman) [reply src/view/frontend/tailwind/tailwind.config.js:10]: phtml is already included with the new mechanism, it also discovers both Tailwind version 2 and 3 and automatically adjusts the purge object.

**Comment by wigman** [comment]
@rajeev-k-tomy Thanks for reviewing!

**Pull request merged**